packages/data-format/index.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-unused-vars */
+import Big from 'big.js'
+Big.RM = 0
+Big.NE = -19
+
+export const fromDripToCfx = numOrStr => {
+  try {
+    const cfx = new Big(numOrStr).div(1e18).toString()
+    return cfx
+  } catch (e) {
+    return numOrStr
+  }
+}
+
+export const fromCfxToDrip = numOrStr => {
+  try {
+    const cfx = new Big(numOrStr).times(1e18).toString()
+    return cfx
+  } catch (e) {
+    return numOrStr
+  }
+}
+
+export const trimZero = numOrStr => {
+  try {
+    return new Big(numOrStr).toString(10)
+  } catch (err) {
+    return numOrStr
+  }
+}
+
+export const formatDigit = (numOrStr, digit) => {
+  try {
+    const str = trimZero(numOrStr)
+    const strArr = str.split('.')
+    // no digit
+    if (!digit) return numOrStr
+    // no decimal or digit < integer length
+    else if (!strArr[1] || strArr[0].length >= digit)
+      return toThousands(strArr[0])
+    else if (strArr[0].length + strArr[1].length <= digit)
+      return toThousands(str)
+    else {
+      const decimal = strArr[1].substring(0, digit - strArr[0].length)
+      return toThousands(trimZero([strArr[0], decimal].join('.')))
+    }
+  } catch (e) {
+    return numOrStr
+  }
+}
+
+export const toThousands = (numOrStr, delimiter = ',', prevDelimiter = ',') => {
+  try {
+    let str = trimZero(numOrStr)
+    return str
+      .replace(new RegExp(prevDelimiter, 'igm'), '')
+      .split('.')
+      .reduce((acc, cur, index) => {
+        if (index) {
+          return `${acc}.${cur}`
+        } else {
+          return cur.replace(/(\d{1,3})(?=(\d{3})+(?:$|\.))/g, `$1${delimiter}`)
+        }
+      }, '')
+  } catch (e) {
+    return numOrStr
+  }
+}
+
+export const formatAmount = numOrStr => {
+  try {
+    const bNum = new Big(numOrStr)
+    const str = trimZero(numOrStr)
+    if (bNum.gte(Big(0)) && bNum.lt(Big(1e6))) {
+      return formatDigit(str, 7)
+    } else if (bNum.gte(Big(1e6)) && bNum.lt(Big(1e9))) {
+      return toThousands(bNum.div(1e6).round(3).toString(10)) + ' M'
+    } else if (bNum.gte(Big(1e9)) && bNum.lt(Big(1e12))) {
+      return toThousands(bNum.div(1e9).round(3).toString(10)) + ' G'
```

Comment:
Don't we need this be `B` as billion? Since wallet is all about assets.

Reply:
https://doc.clickup.com/d/h/3hg1c-6487/2564148bb0983af/3hg1c-1209 we can discuss with Yansong to talk about B or G

Reply:
done, use `B` instead of `G`

Reply:
use `G` finally